### PR TITLE
remove ScriptOutput and related functions

### LINKED ISSUE #652: Deprecate `ScriptOutput`
There doesn't seem to be much reason for this, we are using `TransactionOutput` inside `UnbalancedTx`, we can also remove any relatedconversion function. Let me know if you have any objections.

### PR DESCRIPTION
**Pull request by pb99u069**
Closes #652.

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Contract/Transaction.purs [hunk 1]
```diff
@@ -274,6 +269,9 @@ import Effect.Class (liftEffect)
 import Effect.Exception (throw)
 import Untagged.Union (asOneOf)
 
+-- | NOTE: `scriptOutputToTransactionOutput` is deprecated. See
```
Comment (klntsky) [inline src/Contract/Transaction.purs:272]: What this comment is for? Please remove

File: CHANGELOG.md [hunk 2]
```diff
@@ -65,6 +65,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### Removed
 - `balanceAndSignTxE`, `balanceAndSignTx`, `balanceAndSignTxs`, `balanceTxWithAddress`, `balanceTxsWithAddress`, `withBalancedAndSignedTx` and `withBalancedAndSignedTxs` from `Contract.Transaction` ([#1053](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1053))
+- `ScriptOutput` is deprecated and therefore not exported by `Contract.Address` anymore. also, `Contract.Transaction` doesn't export `scriptOutputToTransactionOutput` anymore ([#1153](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1153)).
```
Comment (klntsky) [inline CHANGELOG.md:68]: Removed, not deprecated.

**Review by klntsky — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (klntsky) [reply CHANGELOG.md:68]: Also: link to the issue and not to the PR please

File: src/Contract/Transaction.purs (hunk 1 above)
Comment (pb99u069) [reply src/Contract/Transaction.purs:272]: sorry, I think I misunderstood Josephs instructions. removed a similar NOTE from `Contract.Address` as well.

File: CHANGELOG.md (hunk 2 above)
Comment (pb99u069) [reply CHANGELOG.md:68]: Until now I always linked to the PR and not to the Issue (as I saw that most changes in the CHANGELOG are). should the default be linking to the issue?

Comment (klntsky) [reply CHANGELOG.md]: They are almost always interlinked, so it's not critical, but the issue (ideally) should explain the motivation for the change better.

**Pull request merged**